packages/jspsych/src/modules/data/DataCollection.ts

```diff
@@ -134,6 +134,17 @@ export class DataCollection {
     return new DataCollection(this.trials.filter(fn));
   }
 
+  filterColumns(columns: Array<string>) {
+    const filtered_data = [];
+    for (const trial of this.trials) {
+      const filtered_columns = Object.fromEntries(
+        columns.filter((key) => key in trial).map((key) => [key, trial[key]])
+      );
+      filtered_data.push(filtered_columns);
+    }
+    return new DataCollection(filtered_data);
```

Comment:
What about an outer `.map()`?
```suggestion
    return new DataCollection(
      this.trials.map((trial) =>
        Object.fromEntries(columns.filter((key) => key in trial).map((key) => [key, trial[key]]))
      )
    );
```